compiler/rustc_hir_typeck/src/writeback.rs

```diff
@@ -748,7 +748,7 @@ impl<'cx, 'tcx> Resolver<'cx, 'tcx> {
                 .infcx
                 .err_ctxt()
                 .emit_inference_failure_err(
-                    Some(self.body.id()),
+                    Some(self.body.id().hir_id.owner.def_id),
```

Comment:
This line is wrong. For a closure, `self.body.id().hir_id.owner.def_id` is the def-id of the enclosing function. You need `body_owner_def_id`.

```suggestion
                    Some(self.tcx.hir().body_owner_def_id(self.body.id())),
```

Reply:
After this change still getting the same errors. I guess the same pattern is happening in other places maybe? ...